tinygrad/codegen/uops.py

```diff
@@ -194,10 +195,12 @@ def loop_collapse(loop_start, loop_end, compval, idx, mval, multconst):
   (UPat(UOps.ALU, BinaryOps.ADD, [UPat(name="x"), UPat(UOps.CONST, 0)]), lambda x: x),   # x+0 -> x or 0+x -> x
   (UPat(UOps.ALU, BinaryOps.MUL, [UPat(name="x"), UPat(UOps.CONST, 1)]), lambda x: x),   # x*1 -> x or 1*x -> x
   (UPat(UOps.ALU, BinaryOps.SUB, (UPat(name="x"), UPat(UOps.CONST, 0))), lambda x: x),   # x-0 -> x
-  (UPat(UOps.ALU, BinaryOps.DIV, (UPat(name="x"), UPat(UOps.CONST, 1))), lambda x: x),   # x/1 -> x
-  (UPat(UOps.ALU, BinaryOps.DIV, (UPat(name="x"), UPat(UOps.CONST, -1))), lambda x: -x), # x/-1 -> -x
+  (UPat(UOps.ALU, BinaryOps.IDIV, (UPat(name="x"), UPat(UOps.CONST, 1))), lambda x: x),   # x/1 -> x
+  (UPat(UOps.ALU, BinaryOps.IDIV, (UPat(name="x"), UPat(UOps.CONST, -1))), lambda x: -x), # x/-1 -> -x
   # ** zero folding **
-  (UPat(UOps.ALU, BinaryOps.MUL, [UPat(), UPat(UOps.CONST, 0, name="c")]), lambda c: c), # x*0 -> 0 or 0*x -> 0
+  #x*0 -> 0 or 0*x -> 0
+  #if x is nan it should render the nan value.
+  (UPat(UOps.ALU, BinaryOps.MUL, [UPat(name="x"), UPat(UOps.CONST, 0, name="c")]), lambda x,c: c if not isinstance(x.arg, (int, float)) else x if math.isnan(x.arg) else c), # noqa: E501
```

Comment:
does `lambda x,c: x if isnan(x.arg) else c),` work?

Reply:
Nop, in some tests `x.arg` is None and `isnan` throws error because expect a real number.

![Screenshot 2024-06-11 at 7 55 07 PM](https://github.com/tinygrad/tinygrad/assets/32320832/3dcb021e-4e39-4cf3-b628-1721914f40fd)

Reply:
hmm that's weird, but probably has nothing to do with this pr